app/src/lib/hostedServices/hostedGitServiceFactory.ts

```diff
@@ -0,0 +1,27 @@
+import { GitHubService } from './github/githubService';
+import { ProjectMetrics } from '$lib/metrics/projectMetrics';
+import { buildContextStore } from '$lib/utils/context';
+import type { RepoInfo } from '$lib/url/gitUrl';
+import type { HostedGitService } from './interface/hostedGitService';
+import type { Octokit } from '@octokit/rest';
+
+// Used on a branch level to acquire the right kind of merge request / checks
+// monitoring service.
+export interface HostedGitServiceFactory {
+	build(repo: RepoInfo): HostedGitService | undefined;
+}
+
+export class DefaultHostedGitServiceFactory implements HostedGitServiceFactory {
+	constructor(private octokit: Octokit | undefined) {}
+
+	build(repo: RepoInfo): HostedGitService | undefined {
+		switch (repo.provider) {
+			case 'github.com':
+				if (!this.octokit) throw new Error('Octokit not available');
+				return new GitHubService(new ProjectMetrics(), this.octokit, repo);
+		}
+	}
+}
+
+export const [getHostedGitServiceFactoryStore, createHostedGitServiceFactoryStore] =
+	buildContextStore<HostedGitServiceFactory | undefined>('HostedGitServiceFactory');
```

Comment:
It looks like we don't make use of this and just depend on the DefaultHostedGitServiceFactory directly in the layout file.

Reply:
My thinking was that if we write a test for e.g. `BranchLane`, then we might not want to use that implementation since it would mean we can't replace any of the services with mocks.

Reply:
@mtsgrd but won't we have the `GithubService` in the store which is an intermediary factory?

Reply:
Right sorry, I mixed things up. Thanks, I'll drop these lines. 👍